contrib/database/sql/sql_test.go

```diff
@@ -190,3 +195,56 @@ func TestMySQLUint64(t *testing.T) {
 	assert.NoError(rows.Err())
 	assert.NoError(rows.Close())
 }
+
+//hangingConnector hangs on Connect until ctx is cancelled.
+type hangingConnector struct{}
+
+func (h *hangingConnector) Connect(ctx context.Context) (driver.Conn, error) {
+	select {
+	case <-ctx.Done():
+		return &panicConn{}, errors.New("context cancelled")
```

Comment:
Why did you use `panicConn`? Why not `nil`?

Reply:
Good question, for some reason I thought that I needed something that implemented the driver type. Returning `nil` is much better and it lets me drop `panicConn` entirely.